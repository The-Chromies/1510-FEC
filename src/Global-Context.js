/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/prop-types */
/* eslint-disable prefer-const */
import React, { useState, createContext } from 'react';
import axios from 'axios';
import uuid from 'node-uuid';
import LocalServer from './env/localServ';

export const ContactContext = createContext();

export const ContactContextProvider = (props) => {
  // global state accessible to all components
  let [productId, setProductId] = useState(18085);
  const [revCount, setRevCount] = useState(0);
  const [avgRating, setAvgRating] = useState(0);
  // arr of products added to outfit list in rc
  const [outfitProduct, setOutfitProduct] = useState([]);
  // arr of styles added to outfit list in rc
  const [outfitStyle, setOutfitStyle] = useState([]);
  // selected product
  const [product, setProduct] = useState(null);
  // selected styles obj
  const [styles, setStyles] = useState(null);
  const localServer = LocalServer;

  // click tracking function
  const clickTracker = (widgetName, e) => {
  // post to /interactions endpoint w/ element's class, widget, & time of click
    let date = new Date();
    let clickData = {
      element: e.target.className,
      widget: widgetName,
      time: date.toTimeString(),
    };
    axios.post(`http://${localServer}:3000/interactions`, clickData)
      .then((res) => {
        // console.log('SUCCESSFUL POST TO INTERACTIONS');
        console.log(res);
      })
      .catch((err) => {
        // console.log('FAILED TO POST TO INTERACTIONS');
        console.log(err);
      });
  };

  const generateStarImage = (ratingValue, keyId) => {
    const starCount = ratingValue > 0 ? (Math.round((ratingValue) * 4) / 4).toFixed(2) : 0;

    let remainder = 0;
    remainder = starCount - remainder;
    const starArr = [];
    for (let j = 0; j < 5; j += 1) {
      switch (remainder) { // 4.25
        case 0.25:
          starArr.push(<div key={uuid()}><img src="public/icons/star/quarterStar.png" alt="" className="star-image" /></div>);
          remainder -= 0.25;
          break;
        case 0.50:
          starArr.push(<div key={uuid()}><img src="public/icons/star/halfStar.png" alt="" className="star-image" /></div>);
          remainder -= 0.50;
          break;
        case 0.75:
          starArr.push(<div key={uuid()}><img src="public/icons/star/threeQuarterStar.png" alt="" className="star-image" /></div>);
          remainder -= 0.75;
          break;
        case 1:
          starArr.push(<div key={uuid()}><img src="public/icons/star/fullStar.png" alt="" className="star-image" /></div>);
          remainder -= 1;
          break;
        case 0:
          starArr.push(<div key={uuid()}><img src="public/icons/star/emptyStar.png" alt="" className="star-image" /></div>);
          break;
        default:
          starArr.push(<div key={uuid()}><img src="public/icons/star/fullStar.png" alt="" className="star-image" /></div>);
          remainder -= 1;
          break;
      }
    }
    return (<div className="inline-star" style={{ display: 'inline-flex' }} key={uuid()}>{starArr}</div>);
  };

  // get selected product & style for ov & rc components
  const getProduct = (id) => {
    axios.get(`http://${localServer}:3000/related/product/${id}`)
      .then((results) => {
        console.log('OUTFITPRODUCT!!', results.data);
        let outfitProductList = [...outfitProduct];
        outfitProductList.push(results.data);
        setOutfitProduct(outfitProductList);
      })
      .catch((err) => {
        console.log('err getting outfit product info:', err);
      });
  };

  const getStyle = (id) => {
    axios.get(`http://${localServer}:3000/overview/styles/${id}`)
      .then((results) => {
        let outfitStyleList = [...outfitStyle];
        outfitStyleList.push(results.data);
        setOutfitStyle(outfitStyleList);
      });
  };

  // add product to outfit list
  const handelAddOutfit = (id) => {
    setProductId(id);
    getProduct(id);
    getStyle(id);
  };

  return (
    <ContactContext.Provider value={{
      productId,
      setProductId,
      generateStarImage,
      revCount,
      setRevCount,
      avgRating,
      product,
      styles,
      setAvgRating,
      clickTracker,
      outfitProduct,
      setOutfitProduct,
      outfitStyle,
      setOutfitStyle,
      getProduct,
      getStyle,
      handelAddOutfit,
      localServer,
    }}
    >
      {props.children}
    </ContactContext.Provider>
  );
};
