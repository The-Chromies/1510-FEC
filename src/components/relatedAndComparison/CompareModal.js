/* eslint-disable react/destructuring-assignment */
import React, { useState, useEffect, useContext } from 'react';
import axios from 'axios';
import { makeStyles } from '@material-ui/core/styles';
import Modal from '@material-ui/core/Modal';
import Backdrop from '@material-ui/core/Backdrop';
import Fade from '@material-ui/core/Fade';
import IconButton from '@material-ui/core/IconButton';
import StarBorderIcon from '@material-ui/icons/StarBorder';
import PropTypes from 'prop-types';
import {
  Navbar, Container, Row, Col, Grid,
} from 'react-bootstrap';

const useStyles = makeStyles((theme) => ({
  modal: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  paper: {
    backgroundColor: theme.palette.background.paper,
    border: '2px solid #000',
    boxShadow: theme.shadows[5],
    padding: theme.spacing(2, 4, 3),
  },
}));

export default function TransitionsModal(props) {
  const classes = useStyles();
  const [open, setOpen] = React.useState(false);

  const handleOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  const getProduct = (id) => {
    axios.get(`http://localhost:3000/related/product/${id}`)
      .then((results) => {
      // console.log('jajajaj', results.data);
      // setRelatedList([results.data, ...relatedList]);
      // setProductInfo(results.data);
      })
      .catch((err) => {
        console.log('err getting product info:', err);
      });
  };

  return (
    <>
      <div>
        <IconButton aria-label="settings">
          <StarBorderIcon onClick={handleOpen} />
        </IconButton>

        <Modal
          aria-labelledby="transition-modal-title"
          aria-describedby="transition-modal-description"
          className={classes.modal}
          open={open}
          onClose={handleClose}
          closeAfterTransition
          BackdropComponent={Backdrop}
          BackdropProps={{
            timeout: 500,
          }}
        >
          <Fade in={open}>
            <div className={classes.paper}>
              <h4 id="transition-modal-title">Comparing</h4>
              <div className="compareTable">
                <table className="table">
                  <thead className="thead-dark">
                    <tr>
                      <th scope="col">Current Product</th>
                      <th scope="col">Features</th>
                      <th scope="col">Related Product</th>
                    </tr>
                  </thead>
                  <tbody>
                    {props.productFeatures.map((featureObj, i) => (
                      <>
                        <tr>
                          <td>current</td>
                          <td>
                            {featureObj.feature}
                          </td>
                          <td className="justify-content-end align-content-end text-*-right">
                            {featureObj.value}
                          </td>
                        </tr>
                      </>
                    ))}
                  </tbody>
                </table>
                {/* <p>{props.productFeatures}</p> */}
              </div>
            </div>
          </Fade>
        </Modal>
      </div>
    </>
  );
}

TransitionsModal.propTypes = {
  // relatedProducts: PropTypes.instanceOf(Array).isRequired,
  // product: PropTypes.instanceOf(Array).isRequired,
  // styles: PropTypes.instanceOf(String).isRequired,
  productFeatures: PropTypes.instanceOf(Array).isRequired,
};
