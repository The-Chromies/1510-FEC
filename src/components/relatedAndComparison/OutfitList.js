/* eslint-disable no-alert */
/* eslint-disable max-len */
/* eslint-disable react/destructuring-assignment */
import React, { useState, useEffect } from 'react';
import { makeStyles } from '@material-ui/core/styles';
// eslint-disable-next-line import/no-extraneous-dependencies
import clsx from 'clsx';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardMedia from '@material-ui/core/CardMedia';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import Collapse from '@material-ui/core/Collapse';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import { red } from '@material-ui/core/colors';
import FavoriteIcon from '@material-ui/icons/Favorite';
import ShareIcon from '@material-ui/icons/Share';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import StarBorderIcon from '@material-ui/icons/StarBorder';
import DeleteForeverIcon from '@material-ui/icons/DeleteForever';
import AddBoxIcon from '@material-ui/icons/AddBox';
import Button from '@material-ui/core/Button';
import PropTypes from 'prop-types';

const useStyles = makeStyles((theme) => ({
  root: {
    maxWidth: 220,
    maxHeight: 380,
    minHeight: 380,
    display: 'inline-block',
  },
  media: {
    height: 0,
    paddingTop: '56.25%', // 16:9
  },
  largeIcon: {
    width: 100,
    height: 100,
    display: 'inline-block',
    alignItems: 'center',
    justifyContent: 'center',
    marginBottom: 350,
    paddingRight: 50,

  },
  bIcon: {
    width: 100,
    height: 100,
    display: 'inline-block',

  },
  expand: {
    transform: 'rotate(0deg)',
    marginLeft: 'auto',
    transition: theme.transitions.create('transform', {
      duration: theme.transitions.duration.shortest,
    }),
  },
  expandOpen: {
    transform: 'rotate(180deg)',
  },

}));

export default function OutfitList(props) {
  // console.log('CARDDDD', props.styles);
  // console.log('HEY', props.styles[0].results)
  const classes = useStyles();
  const [expanded, setExpanded] = React.useState(false);

  const handleExpandClick = () => {
    setExpanded(!expanded);
  };

  const handleCardClick = () => {
    //* ****must route to overview with id of selected product*****
    console.log('CLICKING CARD');
  };

  const handelAddOutfit = () => {
    alert('You added to your outfit!');
  };

  return (
    <>
      <div className="outfitContainer">
        <IconButton className={classes.largeIcon}>
          Add Outfit!
          <AddBoxIcon className={classes.bIcon} onClick={handelAddOutfit} />
        </IconButton>

        <Card className={classes.root}>
          <CardHeader
            className="cardHeader"
            action={
              <DeleteForeverIcon />
        }
        // title={props.product.category}
        // subheader={props.product.category}
            subheader="Product Category"
          />
          <CardMedia
            onClick={handleCardClick}
            className={classes.media}
            image={props.styles}
        // title={props.product.name}
            title="Product Name"
          />
          <CardContent onClick={handleCardClick}>
            <div className="cardInfo" variant="body2" color="textSecondary">
              <h4>Product Name</h4>
              {'Product Description'.substring(0, 100).concat('...')}
              <h5>default Price</h5>
            </div>
          </CardContent>
          <CardActions disableSpacing>
            <IconButton aria-label="add to favorites">
              <FavoriteIcon />
            </IconButton>
            <IconButton aria-label="share">
              <ShareIcon />
            </IconButton>
            <IconButton
              className={clsx(classes.expand, {
                [classes.expandOpen]: expanded,
              })}
              onClick={handleExpandClick}
              aria-expanded={expanded}
              aria-label="show more"
            >
              <ExpandMoreIcon />
            </IconButton>
          </CardActions>
          <Collapse in={expanded} timeout="auto" unmountOnExit>
            <CardContent>
              <Typography paragraph>Method:</Typography>
              <Typography paragraph>
                Heat 1/2 cup of the broth in a pot until simmering, add saffron and set aside for 10
                minutes.
              </Typography>
              <Typography paragraph>
                Heat oil in a (14- to 16-inch) paella pan or a large, deep skillet over medium-high
                heat. Add chicken, shrimp and chorizo, and cook, stirring occasionally until lightly
                browned, 6 to 8 minutes. Transfer shrimp to a large plate and set aside, leaving chicken
                and chorizo in the pan. Add pimentón, bay leaves, garlic, tomatoes, onion, salt and
                pepper, and cook, stirring often until thickened and fragrant, about 10 minutes. Add
                saffron broth and remaining 4 1/2 cups chicken broth; bring to a boil.
              </Typography>
              <Typography paragraph>
                Add rice and stir very gently to distribute. Top with artichokes and peppers, and cook
                without stirring, until most of the liquid is absorbed, 15 to 18 minutes. Reduce heat to
                medium-low, add reserved shrimp and mussels, tucking them down into the rice, and cook
                again without stirring, until mussels have opened and rice is just tender, 5 to 7
                minutes more. (Discard any mussels that don’t open.)
              </Typography>
              <Typography>
                Set aside off of the heat to let rest for 10 minutes, and then serve.
              </Typography>
            </CardContent>
          </Collapse>
        </Card>
      </div>
    </>
  );
}

OutfitList.propTypes = {
  // relatedProducts: PropTypes.instanceOf(Array).isRequired,
  styles: PropTypes.instanceOf(Array).isRequired,
};
