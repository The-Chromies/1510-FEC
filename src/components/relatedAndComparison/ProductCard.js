/* eslint-disable react/destructuring-assignment */
/* eslint-disable import/extensions */

import React, { useState, useEffect, useContext } from 'react';
import { makeStyles } from '@material-ui/core/styles';
// eslint-disable-next-line import/no-extraneous-dependencies
import clsx from 'clsx';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardMedia from '@material-ui/core/CardMedia';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import Collapse from '@material-ui/core/Collapse';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import { red } from '@material-ui/core/colors';
import FavoriteIcon from '@material-ui/icons/Favorite';
import ShareIcon from '@material-ui/icons/Share';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import StarBorderIcon from '@material-ui/icons/StarBorder';
import PropTypes from 'prop-types';
import CompareModal from './CompareModal.js';
import { ContactContext } from '../../Global-Context';

// import Carousel from 'react-multi-carousel';
// import { Paper, Button } from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
  root: {
    maxWidth: 220,
    maxHeight: 380,
    minHeight: 380,
    display: 'inline-block',
  },
  media: {
    height: 0,
    paddingTop: '56.25%', // 16:9
  },

}));

export default function ProductCardRelated(props) {
  const {
    productId, setProductId,
  } = useContext(ContactContext);
  // console.log('CARDDDD', props.styles);
  // console.log('HEY', props.styles[0].results)
  const classes = useStyles();
  // const [expanded, setExpanded] = React.useState(false);

  // const handleExpandClick = () => {
  //   setExpanded(!expanded);
  // };

  const handleCardClick = (id) => {
    setProductId(id);
    console.log('CLICKING CARD');
  };
  console.log('jahdisagdaydg', productId);

  return (
    <>
      <Card className={classes.root}>
        <CardHeader
          className="cardHeader"
          action={
            <CompareModal productFeatures={props.productFeatures} />
        }
        // title={props.product.category}
          subheader={props.product.category}
        />
        <CardMedia
          onClick={() => { handleCardClick(props.product.id); }}
          className={classes.media}
          image={props.styles}
          title={props.product.name}
        />
        <CardContent onClick={() => { handleCardClick(props.product.id); }}>
          <div className="cardInfo" variant="body2" color="textSecondary">
            <h4>{props.product.name}</h4>
            {props.product.description.substring(0, 100).concat('...')}
            <h5>{props.product.default_price}</h5>
            <div>{props.stars(props.rating)}</div>
          </div>
        </CardContent>
        <IconButton aria-label="add to favorites">
          <FavoriteIcon />
        </IconButton>
        <IconButton aria-label="share">
          <ShareIcon />
        </IconButton>
      </Card>
    </>
  );
}

ProductCardRelated.propTypes = {
  // relatedProducts: PropTypes.instanceOf(Array).isRequired,
  product: PropTypes.instanceOf(Array).isRequired,
  styles: PropTypes.instanceOf(String).isRequired,
  productFeatures: PropTypes.instanceOf(Object).isRequired,
  rating: PropTypes.instanceOf(Array).isRequired,
  stars: PropTypes.instanceOf(Function).isRequired,
};

// action={
//   <IconButton aria-label="settings">
//     <CompareModal onClick={handleOpen}/>
//     <StarBorderIcon />
//   </IconButton>
// }
