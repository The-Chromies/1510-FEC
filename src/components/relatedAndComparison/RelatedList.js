/* eslint-disable react/no-array-index-key */
/* eslint-disable max-len */
/* eslint-disable react/destructuring-assignment */
import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import Carousel from 'react-multi-carousel';
import 'react-multi-carousel/lib/styles.css';
import { Paper, Button } from '@material-ui/core';
import ProductCard from './ProductCard';
import OutfitList from './OutfitList';

const responsive = {
  // RULES FOR DISPLAYING CARDS ON SCREEN
  superLargeDesktop: {
    breakpoint: { max: 4000, min: 3000 },
    items: 4,
  },
  desktop: {
    breakpoint: { max: 3000, min: 1024 },
    items: 4,
  },
  tablet: {
    breakpoint: { max: 1024, min: 464 },
    items: 3,
  },
  mobile: {
    breakpoint: { max: 464, min: 0 },
    items: 1,
  },
};
// MAPPING THROUGH RELATED LIST TO DISPLAY RELATED LIST CAROUSEL AND INFO
function RelatedList(props) {
  console.log('test in relatedList', props.productInfo[0].features);
  return (
    <Carousel
      className="relatedCarousel"
      responsive={responsive}
    >
      {props.productInfo.map((product, i) => (
        <ProductCard product={product} key={i} styles={props.styles[i].results[0].photos[0].thumbnail_url} productFeatures={product.features} rating={props.rating} stars={props.stars} />
      ))}
    </Carousel>
  );
}

RelatedList.propTypes = {
  productInfo: PropTypes.instanceOf(Array).isRequired,
  styles: PropTypes.instanceOf(Array).isRequired,
  rating: PropTypes.instanceOf(Array).isRequired,
  stars: PropTypes.instanceOf(Function).isRequired,

};

export default RelatedList;
