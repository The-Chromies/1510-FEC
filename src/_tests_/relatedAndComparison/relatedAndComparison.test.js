import { render, screen } from '@testing-library/react';
import React from 'react';
import Enzyme, { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import RelatedAndComparison from '../../components/relatedAndComparison/rcMain';
// import RatingsAndReviews from '../components/ratingsAndReviews/rcMain';
import { ContactContextProvider } from '../../Global-Context';
import RelatedList from '../../components/relatedAndComparison/RelatedList';
import ProductCard from '../../components/relatedAndComparison/ProductCard';
import OutfitList from '../../components/relatedAndComparison/OutfitList';
import CompareModal from '../../components/relatedAndComparison/CompareModal';

it('matches the RelatedAndComparison snapshot', () => {
  const snapshot = renderer
    .create(<ContactContextProvider><RelatedAndComparison /></ContactContextProvider>)
    .toJSON();
  expect(snapshot).toMatchSnapshot();
});

Enzyme.configure({ adapter: new Adapter() });

describe('RelatedAndComparison', () => {
  let wrapper;
  it('Contains a RelatedAndComparison component', () => {
    wrapper = shallow(<ContactContextProvider><RelatedAndComparison /></ContactContextProvider>);
    expect(wrapper.find()).toBe(true);
  });

  it('Contains a Related List component', () => {
    expect(wrapper.find(<RelatedList />)).toBe(true);
  });

  it('Contains a Product Card component', () => {
    expect(wrapper.find(<ProductCard />)).toBe(true);
  });

  it('Contains an Outfit List component', () => {
    expect(wrapper.find(<OutfitList />)).toBe(true);
  });

  it('Contains a Compare Modal component', () => {
    expect(wrapper.find(<CompareModal />)).toBe(true);
  });
});
