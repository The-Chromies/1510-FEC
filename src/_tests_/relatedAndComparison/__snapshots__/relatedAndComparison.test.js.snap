// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the RelatedAndComparison snapshot 1`] = `
<div
  className="related-comparison-container"
>
  Related And Comparison
  <div
    className="outfitContainer"
  >
    <button
      className="MuiButtonBase-root MuiIconButton-root makeStyles-largeIcon-3"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        Add Outfit!
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root makeStyles-bIcon-4"
          focusable="false"
          onClick={[Function]}
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
          />
        </svg>
      </span>
    </button>
    <div
      className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiCardHeader-root cardHeader"
      >
        <div
          className="MuiCardHeader-content"
        >
          <span
            className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            Product Category
          </span>
        </div>
        <div
          className="MuiCardHeader-action"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </div>
      </div>
      <div
        className="MuiCardMedia-root makeStyles-media-2"
        onClick={[Function]}
        title="Product Name"
      />
      <div
        className="MuiCardContent-root"
        onClick={[Function]}
      >
        <div
          className="cardInfo"
          color="textSecondary"
          variant="body2"
        >
          <h4>
            Product Name
          </h4>
          Product Description...
          <h5>
            default Price
          </h5>
        </div>
      </div>
      <div
        className="MuiCardActions-root"
      >
        <button
          aria-label="add to favorites"
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
              />
            </svg>
          </span>
        </button>
        <button
          aria-label="share"
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z"
              />
            </svg>
          </span>
        </button>
        <button
          aria-expanded={false}
          aria-label="show more"
          className="MuiButtonBase-root MuiIconButton-root makeStyles-expand-5"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
